### PR TITLE
Fix pegout creation index check, now it validates result against kecckak256 zero hash

### PR DESCRIPTION
**Pull request by nathanieliov**
Fix pegout creation index check, now it validates result against kecckak256 zero hash.
Add more tests. Update some comments.

### CODE REVIEW

File: src/test/java/co/rsk/federate/FederatorSupportTest.java [hunk 1]
```diff
@@ -153,16 +162,20 @@ public void getPegoutCreationRskTxHashByBtcTxHash_not_found() {
...
             bridgeTransactionSender
-        );
+        ));
 
         FederationMember federationMember = federation.getMembers().get(0);
         RskAddress rskAddress = new RskAddress(federationMember.getRskPublicKey().getAddress());
         federatorSupport.setMember(federationMember);
 
+        ActivationConfig.ForBlock activationConfig = mock(ActivationConfig.ForBlock.class);
+        doReturn(true).when(activationConfig).isActive(ConsensusRule.RSKIP298);
```
Comment (marcos-iov) [inline src/test/java/co/rsk/federate/FederatorSupportTest.java:176]: Shouldn't we also test pre RSKIP activation?

File: src/test/java/co/rsk/federate/FederatorSupportTest.java [hunk 2]
```diff
@@ -140,7 +141,15 @@ public void getPegoutCreationRskTxHashByBtcTxHash_zero_hash() {
...
-        test_getPegoutCreationRskTxHashByBtcTxHash(sha256Hash, hash, true);
+        test_getPegoutCreationRskTxHashByBtcTxHash(sha256Hash, hash, false);
+    }
+
+    @Test
+    public void getPegoutCreationRskTxHashByBtcTxHash_keccak256_from_a_zero_byte_array() {
+        co.rsk.bitcoinj.core.Sha256Hash sha256Hash = co.rsk.bitcoinj.core.Sha256Hash.ZERO_HASH;
+        Keccak256 hash = new Keccak256(new byte[32]);
+
+        test_getPegoutCreationRskTxHashByBtcTxHash(sha256Hash, hash, false);
```
Comment (marcos-iov) [inline src/test/java/co/rsk/federate/FederatorSupportTest.java:152]: If exists == false, FederatorSupport mock will always return `Keccak256.ZERO_HASH.getBytes()`. So this test is pretty much the same as the one above, it doesn't really matter what value you use for the rsk hash

File: src/test/java/co/rsk/federate/btcreleaseclient/BtcReleaseClientServiceImplTest.java [hunk 3]
```diff
@@ -34,6 +31,13 @@ public void setUp() throws Exception {
         federation = TestUtils.createFederation(params, 1);
     }
 
+    @Test
+    public void getPegoutCreationRskTxHashByBtcTxHash_ok_not_found_in_pegout_creation_index_before_RSKIP298_activation() {
```
[the remark this answers is not in the document]
Comment (nathanieliov) [reply src/test/java/co/rsk/federate/btcreleaseclient/BtcReleaseClientServiceImplTest.java:35]: This is wrong. I'll remove it.

File: src/test/java/co/rsk/federate/btcreleaseclient/BtcReleaseClientServiceImplTest.java [hunk 4]
```diff
@@ -48,6 +52,20 @@ public void getPegoutCreationRskTxHashByBtcTxHash_ok_found_in_legacy_storage() {
...
+        Keccak256 hash = Keccak256.ZERO_HASH;
+        test_getPegoutCreationRskTxHashByBtcTxHash( sha256Hash, hash, false, false);
+    }
+
+    @Test
+    public void getPegoutCreationRskTxHashByBtcTxHash_keccak256_from_a_zero_byte_array() {
+        Sha256Hash sha256Hash = TestUtils.createBtcTransaction(params, federation).getHash();
+        Keccak256 hash = new Keccak256(new byte[32]);
+        test_getPegoutCreationRskTxHashByBtcTxHash( sha256Hash, hash, false, false);
+    }
```
[the remark this answers is not in the document]
Comment (nathanieliov) [reply src/test/java/co/rsk/federate/btcreleaseclient/BtcReleaseClientServiceImplTest.java:67]: No, these tests also show that a Keccak256 ZERO HASH and Keccak256 created from an empty 32-byte array, are being handled as a not found response from the bridge. Basically, these tests cover the scenario that was causing the issue on the validation. This is why I think it'll be good to keep them.

File: src/test/java/co/rsk/federate/FederatorSupportTest.java (hunk 2 above)
Comment (nathanieliov) [reply src/test/java/co/rsk/federate/FederatorSupportTest.java:152]: Ohhh, you are right. I'll try to change that. Thanks!

File: src/test/java/co/rsk/federate/FederatorSupportTest.java [hunk 5]
```diff
@@ -128,51 +129,67 @@ public void hasBlockCoinbaseInformed() {
...
+        // getPegoutCreationRskTxHashByBtcTxHash bridge method returns an empty 32-byte array when no entry is found
+        Keccak256 hash = new Keccak256(new byte[32]);
 
-        test_getPegoutCreationRskTxHashByBtcTxHash(sha256Hash, hash, true);
+        test_getPegoutCreationRskTxHashByBtcTxHash(true, sha256Hash, hash, false);
     }
 
     @Test
-    public void getPegoutCreationRskTxHashByBtcTxHash_not_found() {
+    public void getPegoutCreationRskTxHashByBtcTxHash_not_found_after_RSKIP293() {
```
[the remark this answers is not in the document]
Comment (nathanieliov) [reply src/test/java/co/rsk/federate/FederatorSupportTest.java]: 🤦‍♂️

File: src/test/java/co/rsk/federate/btcreleaseclient/BtcReleaseClientServiceImplTest.java (hunk 4 above)
[the remark this answers is not in the document]
Comment (nathanieliov) [reply src/test/java/co/rsk/federate/btcreleaseclient/BtcReleaseClientServiceImplTest.java]: Resolved.

**Pull request merged**